src/ur/workday.urs

```diff
@@ -0,0 +1,150 @@
+signature AUTH = sig
+    val token : transaction (option string)
+end
+
+(* Nonopaque type for all Workday IDs. *)
+type wid = string
```

Comment:
Why not keep this type abstract?  Maybe the design pattern you were missing is declaring all the relevant type-class instances for the abstract type?  You can find plenty of examples in other World modules.

Reply:
Thanks for the pointer! I didn't realize it was possible to do this.